CHANGELOG.md

```diff
@@ -1,5 +1,12 @@
 # Smithy Gradle Plugin Changelog
 
+## 1.0.1 (2024-07-16)
+
+### Features 
+* Updated version sync script to handle newly added `smithy-trait-package` plugin. ([#142](https://github.com/smithy-lang/smithy-gradle-plugin/pull/142))
```

Comment:
Like with the smithy repo, I don't think we need changelog entries for maintenance items